tst/EnergyPlus/unit/PhaseChangeModeling/HysteresisModel.unit.cc

```diff
@@ -0,0 +1,677 @@
+		previous_temp = updated_temp;
+		if (dir_counter == 1) {
+			if (dir_a_delta_temp > 0 && previous_temp > dir_a_ending_temp) dir_counter++;
+			if (dir_a_delta_temp < 0 && previous_temp < dir_a_ending_temp) dir_counter++;
+		} else {
+			if (dir_b_delta_temp > 0 && previous_temp > dir_b_ending_temp) break;
+			if (dir_b_delta_temp < 0 && previous_temp < dir_b_ending_temp) break;
+		}
+
+	}
+	plot_file.close();
+
+	std::ofstream plot_script_file;
+	plot_script_file.open ("make_plot.plt");
+	plot_script_file << "set terminal dumb\n";
+	plot_script_file << "set datafile separator \",\"\n";
+	plot_script_file << "set title \"Specific Heat = f(Temperature)\"\n";
+	plot_script_file << "plot \"hysteresis_plot.csv\" every ::2 using 1:2 title \"Specific Heat = f(T)\"\n";
+	plot_script_file << "set title \"Specific Heat = f(Time)\"\n";
+	plot_script_file << "plot \"hysteresis_plot.csv\" every ::2 using 0:2 title \"Specific Heat = f(t)\"\n";
+
+	plot_script_file.close();
+	system( "gnuplot make_plot.plt > plot_output.txt" );
+
+	std::ofstream code_file;
+	code_file.open ("hysteresis_code.cpp");
+	code_file << "// Generated by this call:\n";
+	code_file << "// generate_test_data_two_directions(this->ModelA, \"" + test_name + "\", " + std::to_string(initial_temp) + ", " + std::to_string(dir_a_delta_temp) + ", " + std::to_string(dir_a_ending_temp) + ", " + std::to_string(dir_b_delta_temp) + ", " + std::to_string(dir_b_ending_temp) + ", " + std::to_string(initial_state) + ");\n";
+	code_file << "TEST_F(HysteresisTest, " << test_name << ")\n";
+	code_file << "{\n";
+	code_file << "\t// TestDescription\n";
+
+	std::string line;
+	std::ifstream myfile ("plot_output.txt");
+	if (myfile.is_open())
+	{
+		while ( getline (myfile, line) )
+		{
+			code_file << "\t// " << line << "\n";
+		}
+		myfile.close();
+	}
+	code_file << "\tstd::vector<GetSpecHeatArgs> args_list;\n";
+	for ( auto & code_line : code_lines )
+	{
+		code_file << code_line;
+	}
+	code_file << "\tfor ( auto & cp_call : args_list ) {\n";
+	code_file << "\t\tint calculated_pcm_state = -99;\n";
+	code_file << "\t\tReal64 calculated_cp = this->ModelA.getCurrentSpecificHeat(cp_call.previousTemperature, cp_call.updatedTemperature, cp_call.temperatureReverse, cp_call.previousPhaseChangeState, calculated_pcm_state);\n";
+	code_file << "\t\tEXPECT_EQ(cp_call.expectedUpdatedPhaseChangeState, calculated_pcm_state);\n";
+	code_file << "\t\tEXPECT_NEAR(cp_call.expectedSpecificHeat, calculated_cp, 1.0);\n";
+	code_file << "\t}\n";
+	code_file << "}\n";
+	code_file.close();
+}
+
+TEST_F(HysteresisTest, GenerateTestData)
+{
+	//generate_test_data(this->ModelA, "StraightUpCurve", 14, 26, 0.5, 2);
+	//generate_test_data(this->ModelA, "StraightDownCurve", 30, 15, -0.5, -2);
+	//generate_test_data_two_directions(this->ModelA, "IntoMeltingAndBackDown", 14, 0.5, 20, -0.5, 14, 2);
+	generate_test_data_two_directions(this->ModelA, "CompletelyThroughMeltingAndBackDown", 14, 0.5, 30, -0.5, 14, 2);
```

Comment:
Do these generate calls still need to be generated by the unit test? Also do they still need to be included with this PR?